flask/app/manage.py

```diff
@@ -264,191 +147,155 @@ def add_dummy_data():
+    if len(db.session.query(MetaDatasetType_DatasetType).all()) != 0:
+        return
+    for e in ["RES", "CES", "WES", "CPS", "RCS", "RDC", "RDE"]:
         db.session.add(
-            models.PipelineDatasets(pipeline_id=1, supported_metadataset_type="Genome")
+            MetaDatasetType_DatasetType(metadataset_type="Exome", dataset_type=e)
         )
-        # exomic datasets map to pipeline_id 2 (CRE)
+    for g in ["RGS", "CGS", "WGS"]:
         db.session.add(
-            models.PipelineDatasets(pipeline_id=2, supported_metadataset_type="Exome")
+            MetaDatasetType_DatasetType(metadataset_type="Genome", dataset_type=g)
         )
-        db.session.commit()
-        print("Added dataset support info for pipelines")
-
-    # add the metadataset type to dataset type mapping
-    if len(db.session.query(models.MetaDatasetType_DatasetType).all()) == 0:
-        for e in ["RES", "CES", "WES", "CPS", "RCS", "RDC", "RDE"]:
-            db.session.add(
-                models.MetaDatasetType_DatasetType(
-                    metadataset_type="Exome", dataset_type=e
-                )
-            )
-        for g in ["RGS", "CGS", "WGS"]:
-            db.session.add(
-                models.MetaDatasetType_DatasetType(
-                    metadataset_type="Genome", dataset_type=g
-                )
+    for o in ["RLM", "RMM", "RTA"]:
+        db.session.add(
+            MetaDatasetType_DatasetType(metadataset_type="Other", dataset_type=o)
+        )
+
+    db.session.add(
+        MetaDatasetType_DatasetType(metadataset_type="RNA", dataset_type="RRS")
+    )
+    db.session.commit()
+    print("Added metadataset_dataset information")
+
+
+@with_appcontext
+def add_data_hierarchies():
+    if len(db.session.query(Family).all()) != 0:
+        return
+    family_code_iter = 2000
+    participant_code_iter = 1
+    # db.session.query(Group).all()
+    for group in Group.query.filter(Group.group_code != "c4r").all():
+        institution = Institution.query.filter_by(
+            institution=group.group_name
+        ).one_or_none()
+        # default to CHEO for unknown
+        if not institution:
+            cheo = "Children's Hospital of Eastern Ontario"
+            institution = Institution.query.filter_by(institution=cheo).one_or_none()
+
+        # create family per group
+        default_family = Family(
+            family_codename=str(family_code_iter), created_by_id=1, updated_by_id=1
+        )
+        # build trio
+        for sex in ["-", "Female", "Male"]:
+            participant = Participant(
+                participant_codename=f"{group.group_code.upper()}{participant_code_iter:04}",
+                institution_id=institution.institution_id,
+                affected=True if sex == "-" else False,
+                participant_type=ParticipantType.Proband
+                if sex == "-"
+                else ParticipantType.Parent,
+                sex=getattr(Sex, random.choice(["Female", "Male"]))
```

Comment:
🎲

Reply:
> Does the @with_appcontext decorator need to go on every function AND the click function?

I think the context needs to be given to only commands registered to the app (as in only the click functions, ie those which would be called through the command line)